test/test_cli.rb

```diff
@@ -269,4 +269,212 @@ def test_new
       assert(files.include?("locales/en.yml"))
     end
   end
+
+  def mock_rspec_docker_commands(
+    cli,
+    verbose:,
+    setup_commands:,
+    rspec_path: "/spec",
+    container_state: nil,
+    headless: false
+  )
+    cli
+      .expects(:execute)
+      .with(
+        command:
+          "docker ps -a --format '{{json .}}' | grep #{DiscourseTheme::Cli::DISCOURSE_TEST_DOCKER_CONTAINER_NAME}",
+      )
+      .returns(
+        (
+          if container_state
+            %({"Names":"#{DiscourseTheme::Cli::DISCOURSE_TEST_DOCKER_CONTAINER_NAME}","State":"#{container_state}"})
+          else
+            ""
+          end
+        ),
+      )
+
+    if setup_commands
+      if container_state
+        cli.expects(:execute).with(
+          command: "docker stop #{DiscourseTheme::Cli::DISCOURSE_TEST_DOCKER_CONTAINER_NAME}",
+        )
+
+        cli.expects(:execute).with(
+          command: "docker rm -f #{DiscourseTheme::Cli::DISCOURSE_TEST_DOCKER_CONTAINER_NAME}",
+        )
+      end
+
+      cli.expects(:execute).with(
+        command: <<~COMMAND.squeeze(" "),
```

Comment:
Seeing how often `squeeze` is used for the command argument, I wonder if it makes sense to move it to `execute`. I believe this is used only when printing the command line anyway, right? I mean, when executing the command it does not matter if there are some extra spaces.

Reply:
Good point :+1: I'm using squeeze mainly to make the mocks in the tests easier to write. I've moved `squeeze` into the `execute` method.